packages/devtools_app/lib/src/screens/inspector_v2/inspector_tree_controller.dart

```diff
@@ -187,78 +183,99 @@ class InspectorTreeController extends DisposableController
   set selection(InspectorTreeNode? node) {
     if (node == _selection) return;
 
-    setState(() {
-      _selection?.selected = false;
-      _selection = node;
-      _selection?.selected = true;
-      final configLocal = config;
-      if (configLocal.onSelectionChange != null) {
-        configLocal.onSelectionChange!();
-      }
-    });
+    _selection?.selected = false;
+    _selection = node;
+    _selection?.selected = true;
+    final configLocal = config;
+    if (configLocal.onSelectionChange != null) {
+      configLocal.onSelectionChange!();
+    }
+    _updateRows();
   }
 
   InspectorTreeNode? get hover => _hover;
   InspectorTreeNode? _hover;
 
   double? lastContentWidth;
 
-  final cachedRows = <InspectorTreeRow?>[];
   InspectorTreeRow? _cachedSelectedRow;
 
   /// All cached rows of the tree.
   ///
-  /// Similar to [cachedRows] but:
+  /// Similar to [rowsInTree] but:
   /// * contains every row in the tree (including collapsed rows)
   /// * items don't change when nodes are expanded or collapsed
   /// * items are populated only when root is changed
   final _searchableCachedRows = <InspectorTreeRow?>[];
 
-  void setSearchTarget(SearchTargetType searchTarget) {
-    _searchTarget = searchTarget;
-    refreshSearchMatches();
+  /// All the rows that should be displayed in the tree.
+  ///
+  /// The rows can be updated with a call to [_updateRows].
+  ValueListenable<List<InspectorTreeRow?>> get rowsInTree => _rowsInTree;
+  final _rowsInTree = ListValueNotifier<InspectorTreeRow?>([]);
+
+  /// Map from node to the index for that node's row in [rowsInTree].
+  final _nodeToRowIndex = <InspectorTreeNode, int>{};
+
+  /// Rebuilds the tree and updates [rowsInTree] with the new values.
+  ///
+  /// If [updateSearchableRows] is true, also updates [_searchableCachedRows]
+  /// with the new values.
+  ///
+  /// TODO(elliette): Consider only updating an [InspectorTreeNode]'s branch
+  /// when it is marked as dirty, instead of the entire tree. See:
+  /// https://github.com/flutter/devtools/issues/7980
```

Comment:
place this outside of the dart doc for the method

Reply:
Done!